kv/kvserver: TestReplicateQueueExpirationLeasesOnly failed

kv/kvserver.TestReplicateQueueExpirationLeasesOnly [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132679?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9132679?buildTab=artifacts#/) on master @ [53dbb86acb1d48309530181b94838faf937084d3](https://github.com/cockroachdb/cockroach/commits/53dbb86acb1d48309530181b94838faf937084d3):


```
=== RUN   TestReplicateQueueExpirationLeasesOnly
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly2009915580
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2419: initial: epochLeases=316 expLeases=2
    replicate_queue_test.go:2426: enabling: epochLeases=300 expLeases=18
    replicate_queue_test.go:2426: enabling: epochLeases=287 expLeases=31
    replicate_queue_test.go:2426: enabling: epochLeases=274 expLeases=44
    replicate_queue_test.go:2426: enabling: epochLeases=260 expLeases=53
    replicate_queue_test.go:2426: enabling: epochLeases=248 expLeases=57
    replicate_queue_test.go:2426: enabling: epochLeases=232 expLeases=61
    replicate_queue_test.go:2426: enabling: epochLeases=221 expLeases=58
    replicate_queue_test.go:2426: enabling: epochLeases=210 expLeases=56
    replicate_queue_test.go:2426: enabling: epochLeases=200 expLeases=51
    replicate_queue_test.go:2426: enabling: epochLeases=185 expLeases=53
    replicate_queue_test.go:2426: enabling: epochLeases=176 expLeases=53
    replicate_queue_test.go:2426: enabling: epochLeases=162 expLeases=60
    replicate_queue_test.go:2426: enabling: epochLeases=150 expLeases=62
    replicate_queue_test.go:2426: enabling: epochLeases=138 expLeases=61
    replicate_queue_test.go:2426: enabling: epochLeases=130 expLeases=56
    replicate_queue_test.go:2426: enabling: epochLeases=121 expLeases=51
    replicate_queue_test.go:2426: enabling: epochLeases=111 expLeases=51
    replicate_queue_test.go:2426: enabling: epochLeases=100 expLeases=55
    replicate_queue_test.go:2426: enabling: epochLeases=92 expLeases=56
    replicate_queue_test.go:2426: enabling: epochLeases=85 expLeases=58
    replicate_queue_test.go:2426: enabling: epochLeases=73 expLeases=62
    replicate_queue_test.go:2426: enabling: epochLeases=62 expLeases=59
    replicate_queue_test.go:2424: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replicate_queue_test.go:2424
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueExpirationLeasesOnly
    replicate_queue_test.go:2426: enabling: epochLeases=52 expLeases=57
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly2009915580
--- FAIL: TestReplicateQueueExpirationLeasesOnly (300.42s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueExpirationLeasesOnly.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25630

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2365,11 +2365,6 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 	skip.UnderRace(t) // too slow under stressrace
 	skip.UnderShort(t)
 
-	timeout := 5 * time.Second
-	if skip.Stress() {
-		timeout = 30 * time.Second
-	}
-
 	ctx := context.Background()
 	st := cluster.MakeTestingClusterSettings()
 	kvserver.ExpirationLeasesOnly.Override(ctx, &st.SV, false) // override metamorphism
@@ -2425,10 +2420,10 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 		epochLeases, expLeases = countLeases()
 		t.Logf("enabling: epochLeases=%d expLeases=%d", epochLeases, expLeases)
 		return epochLeases == 0 && expLeases > 0
-	}, timeout, 500*time.Millisecond)
+	}, 30*time.Second, 500*time.Millisecond) // accomodate stress/deadlock builds
 
 	// Run a scan across the ranges, just to make sure they work.
-	scanCtx, cancel := context.WithTimeout(ctx, timeout)
+	scanCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
 	defer cancel()
 	_, err = db.Scan(scanCtx, scratchKey, scratchKey.PrefixEnd(), 1)
 	require.NoError(t, err)
@@ -2444,5 +2439,5 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 		epochLeases, expLeases = countLeases()
 		t.Logf("disabling: epochLeases=%d expLeases=%d", epochLeases, expLeases)
 		return epochLeases > 0 && expLeases > 0 && expLeases <= initialExpLeases
-	}, timeout, 500*time.Millisecond)
+	}, 30*time.Second, 500*time.Millisecond)
 }

